app/controllers/tasks_controller.rb

```diff
@@ -1,12 +1,23 @@
 class TasksController < ApplicationController
   before_action :require_user
   before_action :set_task, only: [:edit, :update, :destroy]
+  before_action :index_params, only: :index
 
   def index
-    @tasks = current_user.tasks.paginate(page: params[:page], per_page: 50)
+    unless params[:search].nil?
+      search = params[:search].blank? ? "" : params[:search]
+      completion = params[:completion] == "true" ? true : params[:completion] == "false" ? false : [true, false]
+      priority = params[:priority].blank? ? [0, 1, 2] : params[:priority].to_i
+      if params[:category_id].blank?
+        @tasks = current_user.tasks.where(completed: completion, priority: priority).where("name LIKE ?", "%#{search}%").paginate(page: params[:page], per_page: 50)
+      else
+        @tasks = current_user.categories.find(params[:category_id].to_i).tasks.where(completed: completion, priority: priority).where("name LIKE ?", "%#{search}%").paginate(page: params[:page], per_page: 50)
```

Comment:
completion = params[:completion] == "true" ? true : params[:completion] == "false" ? false : [true, false]
priority = params[:priority].blank? ? [0, 1, 2] : params[:priority].to_i

@tasks = current_user.tasks

@tasks = @tasks.joins(:task_categories).where(task_categories: {category_id: params[:category_id].to_i}) if params[:category_id].present?
@tasks = @tasks.where("name LIKE ?", "%#{search}%") if params[:search].present?
@tasks = @tasks.where(completed: completion) if params[: completion].present?
@tasks = @tasks.where(priority: priority) if params[: priority].present?

@tasks = @tasks.paginate(page: params[:page], per_page: 50)

Reply:
searchでやり方を検索するとgetの方が出てきます。（postだと動かないです）
get でやっている所で以下の説明をみました。
https://medium.com/@yassimortensen/simple-search-form-in-rails-8483739e4042

When submitting the search, the form will load the superheroes_path url and submit a GET request. This is because we want to filter the results on our index page, not post anything to our database. After clicking submit, the program will reload the index view page, but this time with a :search param.